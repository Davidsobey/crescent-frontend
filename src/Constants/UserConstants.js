const UserConstants = {
  REGISTER_REQUEST: 'USERS_REGISTER_REQUEST',
  REGISTER_SUCCESS: 'USERS_REGISTER_SUCCESS',
  REGISTER_FAILURE: 'USERS_REGISTER_FAILURE',

  LOGIN_REQUEST: 'USERS_LOGIN_REQUEST',
  LOGIN_SUCCESS: 'USERS_LOGIN_SUCCESS',
  LOGIN_FAILURE: 'USERS_LOGIN_FAILURE',
  LOGIN_CLOSE: 'USERS_LOGIN_CLOSE',

  LOGOUT: 'USERS_LOGOUT',

  GETALL_REQUEST: 'USERS_GETALL_REQUEST',
  GETALL_SUCCESS: 'USERS_GETALL_SUCCESS',
  GETALL_FAILURE: 'USERS_GETALL_FAILURE',

  GET_ALL_ROLES_REQUEST: 'GET_ALL_ROLES_REQUEST',
  GET_ALL_ROLES_SUCCESS: 'GET_ALL_ROLES_SUCCESS',
  GET_ALL_ROLES_FAILURE: 'GET_ALL_ROLES_FAILURE',

  GETUSER_REQUEST: 'USERS_GETUSER_REQUEST',
  GETUSER_SUCCESS: 'USERS_GETUSER_SUCCESS',
  GETUSER_FAILURE: 'USERS_GETUSER_FAILURE',

  DELETE_REQUEST: 'USERS_DELETE_REQUEST',
  DELETE_SUCCESS: 'USERS_DELETE_SUCCESS',
  DELETE_FAILURE: 'USERS_DELETE_FAILURE',

  EDIT_USER_REQUEST: 'USERS_EDIT_REQUEST',
  EDIT_USER_SUCCESS: 'USERS_EDIT_SUCCESS',
  EDIT_USER_FAILURE: 'USERS_EDIT_FAILURE',

  ENROL_REQUEST: 'ENROL_USER_REQUEST',
  ENROL_SUCCESS: 'ENROL_USER_SUCCESS',
  ENROL_FAILURE: 'ENROL_USER_FAILURE',

  ENROL_ALL_REQUEST: 'ENROL_ALL_REQUEST',
  ENROL_ALL_SUCCESS: 'ENROL_ALL_SUCCESS',
  ENROL_ALL_FAILURE: 'ENROL_ALL_FAILURE',

  EDIT_USER_SUCCESS: 'EDIT_USER_SUCCESS',
};

export default UserConstants;
