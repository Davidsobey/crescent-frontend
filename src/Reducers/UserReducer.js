import UserConstants from '../Constants/UserConstants';

function UserReducer(state = {}, action) {
  switch (action.type) {
    case UserConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case UserConstants.GETALL_SUCCESS:
      return {
        users: action.users,
      };
    case UserConstants.GETALL_FAILURE:
      return {
        loading: false,
        error: action.error,
      };
    case UserConstants.REGISTER_REQUEST:
      return {
        users: action.userName,
      };
    case UserConstants.REGISTER_SUCCESS:
      return {
        users: action.userName,
      };
    case UserConstants.REGISTER_FAILURE:
      return state;
    case UserConstants.ENROL_REQUEST:
      return {
        enrolments: action.enrolment,
      };
    case UserConstants.ENROL_SUCCESS:
      return {
        enrolments: action.enrolment,
      };
    case UserConstants.ENROL_FAILURE:
      return state;
    case UserConstants.GETUSER_REQUEST:
      return {
        loading: true,
      };
    case UserConstants.GETUSER_SUCCESS:
      return Object.assign({}, state, {
        user: action.user,
      });
    case UserConstants.GETUSER_FAILURE:
      return {
        error: action.error,
      };

    default:
      return state;
  }
}

export default UserReducer;
