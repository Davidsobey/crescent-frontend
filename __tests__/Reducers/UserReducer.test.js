// Reducer to be tested
import UserReducer from '../../src/Reducers/UserReducer';
import UserConstants from '../../src/Constants/UserConstants';

describe('UserReducer', () => {
  test('GETALL_SUCCESS', () => {
    const action = { 
      type: UserConstants.GETALL_SUCCESS, 
      users: [],
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });

  test('GET_ALL_ROLES_SUCCESS', () => {
    const action = { 
      type: UserConstants.GET_ALL_ROLES_SUCCESS,
      roles: [],
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });

  test('REGISTER_SUCCESS', () => {
    const action = { 
      type: UserConstants.REGISTER_SUCCESS,
      userName: "a",
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });

  test('ENROL_SUCCESS', () => {
    const action = { 
      type: UserConstants.ENROL_SUCCESS,
      enrolment: "enrolment",
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETUSER_SUCCESS', () => {
    const action = { 
      type: UserConstants.GETUSER_SUCCESS,
      id: 0,
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });

  test('DELETE_SUCCESS', () => {
    const action = { 
      type: UserConstants.DELETE_SUCCESS,
      id: 0,
    };
    const state = { 
      users: [],
    };

    expect(UserReducer(state, action)).toMatchSnapshot();
  });

  test('EDIT_USER_SUCCESS', () => {
    const action = { 
      type: UserConstants.EDIT_USER_SUCCESS,
      user: undefined,
      user_editing: false,
    };

    expect(UserReducer(undefined, action)).toMatchSnapshot();
  });
});
