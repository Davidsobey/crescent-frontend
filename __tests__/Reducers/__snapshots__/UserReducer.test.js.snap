// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserReducer DELETE_SUCCESS 1`] = `
Object {
  "loading": false,
  "users": Array [],
}
`;

exports[`UserReducer EDIT_USER_SUCCESS 1`] = `
Object {
  "user": undefined,
  "user_editing": false,
}
`;

exports[`UserReducer ENROL_SUCCESS 1`] = `
Object {
  "enrolling": false,
  "enrolment": "enrolment",
}
`;

exports[`UserReducer GET_ALL_ROLES_SUCCESS 1`] = `
Object {
  "loading": false,
  "roles": Array [],
}
`;

exports[`UserReducer GETALL_SUCCESS 1`] = `
Object {
  "loading": false,
  "users": Array [],
}
`;

exports[`UserReducer GETUSER_SUCCESS 1`] = `
Object {
  "user": undefined,
  "user_loading": false,
}
`;

exports[`UserReducer REGISTER_SUCCESS 1`] = `
Object {
  "creating": false,
  "users": "a",
}
`;
